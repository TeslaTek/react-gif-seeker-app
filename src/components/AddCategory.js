import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) { 
            setCategories( cats => [inputValue, ...cats]);
            setinputValue('');
        }
       
        
    }
    return (
        <>
        <form onSubmit={handleSubmit}>
            <input
                id='textBox'
                type="text" 
                placeholder="Agregar categoria"  
                value={inputValue}
                onChange={handleInputChange}
            />
            
        </form>
        <button onClick={handleSubmit}>Search</button>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}