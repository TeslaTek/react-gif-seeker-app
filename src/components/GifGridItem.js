import React from 'react'

export  const  GifGridItem = ( {title, url}) => {
    return (
        <div  className="gifContainer">
            <img src={url} width="300" alt={title}/> 
            <h4>{title} </h4>                        
        </div>
    )
}
