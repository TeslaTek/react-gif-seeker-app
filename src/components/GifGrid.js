import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category}) => {
   


    
   const {data:images, loading} =  useFetchGifs(category);



    return (
        <>
            <div className='animate__animated animate__fadeIn'>
                <h3>{category}</h3>
                <hr/>
            </div>
            { loading && <p className='animate__animated animate__flash' >Loading ...</p>}
            <section className='gridContainer  animate__animated animate__fadeIn '>
                {   images.map( img => (
                         <GifGridItem key={img.id} {...img} />
                ))
                }
            </section>           
        </>
    )
}
