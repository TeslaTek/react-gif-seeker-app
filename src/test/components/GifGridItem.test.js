import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem'

describe('testing GifGridItem component', () => {
    const title = 'xuxa';
    const url = 'https://media0.giphy.com/media/1xVfKGUiAh0TWrZkN9/giphy.gif?cid=be4fc6a44gfvd9765lcax753gy52pjo86o1e68fkxv3w68ju&rid=giphy.gif&ct=g'

    const wrapper = shallow(<GifGridItem title = {title} url = {url} />);
   
    test('should show the gif card', () => {            

        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener titulo', () => {
         const h4 = wrapper.find('h4');
         expect(h4.text().trim()).toBe(title);

    })
    
    test('imagen con url', () => {
        const img = wrapper.find('img');
        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt')).toBe(title)
    })
    
    test('clases', () => {

        const div = wrapper.find('div');
        const classList = div.prop('className');
        expect( div.hasClass('gifContainer')).toEqual(true);

        expect( classList.includes('gifContainer')).toBe(true)
        
    })
    
    
    
})
