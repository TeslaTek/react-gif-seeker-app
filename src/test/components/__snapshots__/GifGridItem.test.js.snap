// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifGridItem component should show the gif card 1`] = `
<div
  className="gifContainer"
>
  <img
    alt="xuxa"
    src="https://media0.giphy.com/media/1xVfKGUiAh0TWrZkN9/giphy.gif?cid=be4fc6a44gfvd9765lcax753gy52pjo86o1e68fkxv3w68ju&rid=giphy.gif&ct=g"
    width="300"
  />
  <h4>
    xuxa
     
  </h4>
</div>
`;
